cmd/bootstrap/cmd/partner_infos.go

```diff
@@ -0,0 +1,244 @@
+		if err != nil {
+			log.Fatal().Err(err).Msgf("could not node info")
+		}
+
+		if isFlowNode(nodePubInfo.Address) {
+			skippedNodes++
+			continue
+		}
+
+		writeNodePubInfoFile(nodePubInfo)
+		partnerStakes[nodePubInfo.NodeID] = defaultPartnerStake
+		numOfNodesByType[nodePubInfo.Role]++
+		totalNumOfPartnerNodes++
+	}
+
+	writePartnerStakesFile(partnerStakes)
+
+	printNodeCounts(numOfNodesByType, totalNumOfPartnerNodes, skippedNodes)
+}
+
+// getFlowClient will validate the flagANNetworkKey and return flow client
+func getFlowClient() *client.Client {
+	// default to insecure client connection
+	insecureClient := true
+
+	if flagANNetworkKey != "" {
+		err := validateANNetworkKey(flagANNetworkKey)
+		if err != nil {
+			log.Fatal().Err(err).Msgf("failed to create flow client invalid access-network-key provided (%s)", flagANNetworkKey, err)
+		}
+
+		insecureClient = false
+	}
+
+	config, err := common.NewFlowClientConfig(flagANAddress, strings.TrimPrefix(flagANNetworkKey, "0x"), insecureClient)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("could not get flow client config with address (%s) and network key (%s)", flagANAddress, flagANNetworkKey)
+	}
+
+	flowClient, err := common.FlowClient(config)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("could not get flow client with address (%s) and network key (%s)", flagANAddress, flagANNetworkKey)
+	}
+
+	return flowClient
+}
+
+// executeGetProposedTableScript executes the get proposed table script
+func executeGetProposedTableScript(ctx context.Context, env templates.Environment, flowClient *client.Client) (cadence.Array, error) {
+	proposedTable, err := flowClient.ExecuteScriptAtLatestBlock(ctx, templates.GenerateReturnProposedTableScript(env), []cadence.Value{})
+	if err != nil {
+		return cadence.Array{}, fmt.Errorf("failed to execute the get proposed table script: %w", err)
+	}
+
+	return proposedTable.(cadence.Array), nil
+}
+
+// executeGetNodeInfoScript executes the get node info script
+func executeGetNodeInfoScript(ctx context.Context, env templates.Environment, client *client.Client, nodeID cadence.Value) (cadence.Value, error) {
+	info, err := client.ExecuteScriptAtLatestBlock(ctx, getNodeInfoScript, []cadence.Value{nodeID})
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute the get node info script: %w", err)
+	}
+
+	return info, nil
+}
+
+// parseNodeInfo convert node info retrieved from
+func parseNodeInfo(info cadence.Value) (*NodePubInfo, error) {
+	fields := info.(cadence.Struct).Fields
+	role := flow.Role(fields[roleField].(cadence.UInt8))
+	return &NodePubInfo{
+		Role:          role.String(),
+		Address:       string(fields[networkingAddressField].(cadence.String)),
+		NodeID:        string(fields[idField].(cadence.String)),
+		Stake:         fields[tokensStakedField].(cadence.UFix64).String(),
+		NetworkPubKey: string(fields[networkingKeyField].(cadence.String)),
+		StakingPubKey: string(fields[stakingKeyField].(cadence.String)),
+	}, nil
```

Comment:
After switching to the existing NodeInfoPub struct, we'll need to parse these differently. 
* `NodeID` is a hex-encoded Identifier
* `NetworkPubKey` and `StakingPubKey` are hex-encoded public keys